ompi/mca/sharedfp/sm/sharedfp_sm_file_open.c

```diff
@@ -135,7 +135,6 @@ int mca_sharedfp_sm_file_open (struct ompi_communicator_t *comm,
         return OMPI_ERROR;
     }
 
-        free(sm_filename);
```

Comment:
Is this `free()` supposed to be part of this pull request?  It doesn't seem related to the semaphore stuff.